tests/jenkins/run_test_installation_docs.sh

```diff
@@ -302,7 +304,7 @@ ubuntu_python_cpu_source()
     set -e
     echo
     echo "### Testing Build From Source ###"
-    echo "${buildfromsource_commands}"
+    buildfromsource_commands=${buildfromsource_commands//cd incubator-mxnet;/cd incubator-mxnet; git checkout ${GIT_SHA}; git submodule update;}
```

Comment:
Jenkins already makes sure you are in the correct git state and hash. Doing a manual change of the git state (e.g. by executing clone or checkout) can result in corruption of the state and thus lead to unexpected result. You can assume that Jenkins checked out the hash that it's supposed to validate. This approach allows us to stay independent of any external resources and fully operate on the git source that the job was requested to run on.

If there's any other git stuff happening during this process (since it's intended for endusers), I'd recommend to filter them as part of this execution.

Reply:
since it's getting git-ceptioned I don't think it would mangle the state. But yeah, maybe filtering any git command would make more sense. Lemme do that!